sdk/core/Azure.Core/src/Shared/OperationInternalBase.cs

```diff
@@ -114,8 +119,21 @@ protected OperationInternalBase(ClientDiagnostics clientDiagnostics, Response ra
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
         /// <returns>The last HTTP response received from the server, including the final result of the long-running operation.</returns>
         /// <exception cref="RequestFailedException">Thrown if there's been any issues during the connection, or if the operation has completed with failures.</exception>
-        public virtual async ValueTask<Response> WaitForCompletionResponseAsync(CancellationToken cancellationToken) =>
-            await WaitForCompletionResponseAsync(DefaultPollingInterval, cancellationToken).ConfigureAwait(false);
+        public virtual async ValueTask<Response> WaitForCompletionResponseAsync(CancellationToken cancellationToken)
```

Comment:
In the case where someone passes in a PollingInterval we need to override the set the PollingStrategy to a new instance IF its currently a LinearPollingStrategy